### PR TITLE
Cleaned folder structure, updated Readme file, added comments to each method in RegionGrowing.py

### PR DESCRIPTION
**Pull request by deeprodge**

### CODE REVIEW

File: unsupervised_rock_detection_3d/README.md [hunk 1]
```diff
@@ -4,10 +4,52 @@
...
+        * Assign labels to points based on the region they belong to.
+4. **Post Processing**:
+
+    * Perform conditional label propagation to assign labels to remaining unlabeled points based on majority labels of their neighbors.
+    * Color the segmented point cloud for visualization by assigning different colors to different regions (e.g., red for rocks, blue for terrain).
+    * Visualize the segmented point cloud to assess the regions' boundaries.
+
+#### Novelty
+
+One significant novelty in our approach is the use of second-order neighbors for smoothness calculation. Our point cloud is densely packed, resulting in neighboring points being very close to one another. This proximity causes the normal orientation differences to appear nearly parallel, leading almost every point to be included in the region. To address this issue, we modified the approach by considering the minimum dot product (maximum deviation) with second-order neighboring points instead of comparing the immediate neighboring points. This adjustment improves the segmentation accuracy by preventing the over-inclusion of points in a region due to near-parallel normals in densely packed clouds.
```
Comment (ZhiangChen) [inline unsupervised_rock_detection_3d/README.md:67]: you should mention the motivation of using second-order neighbors in region growing

File: unsupervised_rock_detection_3d/README.md [hunk 2]
```diff
@@ -4,10 +4,52 @@
...
+    * Identify the lowest point in the bounding box, representing a potential terrain seed.
+
+3. **Region Growing**:
+
+    * Precompute neighbors for each point within the specified distance threshold to speed up segmentation.
+    * Initialize regions from the selected seed points (highest point for rocks, bottommost point for terrain).
+    * Expand regions from seeds by evaluating neighbors based on the segmentation criteria (normal vector smoothness and curvature):
+        * Start with the seed point in a queue.
+        * For each point, evaluate its neighbors based on the segmentation criteria.
+        * Check if neighbors meet the smoothness and curvature thresholds. If they do, add them to the current region and the queue.
```
Comment (ZhiangChen) [inline unsupervised_rock_detection_3d/README.md:38]: you should add the mathematical definition of smoothness and curvature. Also, please clarify that we should just use one of them.

File: unsupervised_rock_detection_3d/README.md (hunk 1 above)
Comment (deeprodge) [reply unsupervised_rock_detection_3d/README.md:67]: I've already discussed the issue we faced and the reason behind using this approach. Is there anything else specific you think should be included or any specific format?

File: unsupervised_rock_detection_3d/README.md (hunk 2 above)
Comment (deeprodge) [reply unsupervised_rock_detection_3d/README.md:38]: Got it! Will add it and create another PR.

File: unsupervised_rock_detection_3d/README.md (hunk 1 above)
Comment (ZhiangChen) [reply unsupervised_rock_detection_3d/README.md:67]: Ah I didn't read carefully. Thanks for clarification. We are good

**Review by ZhiangChen — APPROVED** [review]

**Pull request merged**